BCStudentSoftwareDevTeam/celts
# 1151 other community experience rework

Resolves #1151

## Issue
In order to request an 'other' experience (for one of four experiences you need to finish the CELTS minor), you need to fill out a specialized form which takes in the following information:
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/ac03e920-dc01-4283-bbc2-88680f869b96)
The previous state of the form didn't have all of these fields.


## Solution
We added more fields that are error-checked and address all of the fields listed in the image.

## To Test
- Load up celts and reset database `from-backup`
- Navigate Admin Minor Management
![image](https://github.com/BCStudentSoftwareDevTeam/celts/assets/89316750/fe06945a-5ac3-47fa-8c4b-8cb1a16e84a0)
- Ensure that each criterion that is mentioned in the above image is requested on the form.
- Ensure that you can't put any erroneous data into email or phone number fields
- Ensure that when saved, all relevant form data is saved into the `communityengagementrequest` database table.
- Also run tests and ensure they pass
